src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -394,20 +481,36 @@ protected int getMinTankCapacity(@Nonnull IMultipleTankHandler tanks) {
         return result;
     }
 
+    /**
+     * Find a recipe using inputs
+     * @param maxVoltage the maximum voltage the recipe can have
+     * @param inputs the item inputs used to search for the recipe
+     * @param fluidInputs the fluid inputs used to search for the recipe
+     * @return the recipe if found, otherwise null
+     */
+    @Nullable
     protected Recipe findRecipe(long maxVoltage, IItemHandlerModifiable inputs, IMultipleTankHandler fluidInputs) {
-
-        if(!isRecipeMapValid(getRecipeMap())) {
+        if (getRecipeMap() == null || !isRecipeMapValid(getRecipeMap())) {
```

Comment:
You should create local variable for recipe map and call `getRecipeMap` only once as it can be time consuming. And then do all checks against local variable.